### PR TITLE
SmallRye CP - replace ManagedExecutor producer with synthetic bean

### LINKED ISSUE #10963: Can't inject ManagedExecutor since quarkus 1.6.0
**Describe the bug**
I'm trying to inject a ManagedExecutor (using Kotlin and injecting into `@Singleton` bean that extends `MutinyExampleServiceGrpc.ExampleServiceImplBase` if that matters) and get 
```
Parameter specified as non-null is null: method com.example.Example.<init>, parameter managedExecutor"
```
I tried both constructor injection as well as field injection via `lateinit var`

**Expected behavior**
ManagedExecutor is injected

**Actual behavior**
ManagedExecutor is not injected to my bean (both 1.6.0.Final and 1.6.1.Final)
But it's injected successfully if I change the version to 1.5.2.Final

### PR DESCRIPTION
**Pull request by mkouba**
- deprecate ManagedExecutorInitializedBuildItem
- resolves #10963

### CODE REVIEW

**Review by geoand — APPROVED** [review]
Great! One more concurrency bug squashed :)

File: extensions/smallrye-context-propagation/runtime/src/main/java/io/quarkus/smallrye/context/runtime/SmallRyeContextPropagationProvider.java [hunk 1]
```diff
@@ -1,53 +1,21 @@
...
 
-import org.eclipse.microprofile.context.ManagedExecutor;
 import org.eclipse.microprofile.context.ThreadContext;
 
 import io.quarkus.arc.DefaultBean;
-import io.smallrye.context.SmallRyeManagedExecutor;
-import io.smallrye.context.SmallRyeThreadContext;
 
-@ApplicationScoped
+@Dependent
```
Comment (gsmet) [inline extensions/smallrye-context-propagation/runtime/src/main/java/io/quarkus/smallrye/context/runtime/SmallRyeContextPropagationProvider.java:11]: What's the rationale for this change? (asking as I would like to understand)

Comment (mkouba) [reply extensions/smallrye-context-propagation/runtime/src/main/java/io/quarkus/smallrye/context/runtime/SmallRyeContextPropagationProvider.java:11]: The producer is `@Singleton`, i.e. will only be called once and the result is stored in the singleton context. Therefore, there is no need to keep the `SmallRyeContextPropagationProvider` instance around. Not to mention the generated client proxy class.

**Review by stuartwdouglas — APPROVED** [review]

**Comment by mkouba** [comment]
Merging as the JDK Java 14 JVM Tests failed 2x with:

```
2020-07-28T03:13:55.2107677Z ##[section]Starting: Request a runner to run this job
2020-07-28T03:13:55.8554892Z Can't find any online and idle self-hosted runner in current repository that matches the required labels: 'ubuntu-latest'
2020-07-28T03:13:55.8554999Z Can't find any online and idle self-hosted runner in current repository's account/organization that matches the required labels: 'ubuntu-latest'
2020-07-28T03:13:55.8555063Z Found online and idle hosted runner in current repository's account/organization that matches the required labels: 'ubuntu-latest'
2020-07-28T03:13:56.0495492Z ##[section]Finishing: Request a runner to run this job
```

**Pull request merged**